### PR TITLE
feat: :art: Add Monsters

### PR DESCRIPTION
**Pull request by RaxCruz**
## Description

[5936](https://redmine.kingkit.codes/issues/5936)

## Motivation and Context

切版- 小怪獸 

## Screenshots (if appropriate)

chrome-capture-2022-9-25 (1)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: my-app/src/MonsterButton/MonsterButton.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      case "blue":
+        return <BlueMonsterImage />;
+      default:
+        return <></>;
+    }
+  };
+  return (
+    <div
+      className={"MonsterButton " + (monsterClick ? "MonsterButton-click" : "")}
+      onClick={() => handleMonsterClick()}
```
Comment (shengyuan) [inline my-app/src/MonsterButton/MonsterButton.tsx]: 這裡可以寫
`onClick={handleMonsterClick}` 
簡潔一些。

File: my-app/src/MonsterButton/MonsterButton.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}
+function MonsterButton({ monsterColor }: Props) {
+  const [monsterClick, setMonsterClick] = useState(false);
+  const handleMonsterClick = () => {
+    setMonsterClick(true);
+    setTimeout(() => setMonsterClick(false), 100);
+  };
+  const renderMonster = (monsterColor: string) => {
+    switch (monsterColor) {
+      case "green":
```
Comment (shengyuan) [inline my-app/src/MonsterButton/MonsterButton.tsx:18]: 建議可以抽成 typescript enum 的寫法。

File: my-app/src/MonsterButton/MonsterButton.tsx (hunk 1 above)
Comment (RaxCruz) [reply my-app/src/MonsterButton/MonsterButton.tsx]: 已調整

File: my-app/src/MonsterButton/MonsterButton.tsx (hunk 2 above)
Comment (RaxCruz) [reply my-app/src/MonsterButton/MonsterButton.tsx:18]: 目前考慮到有 props 傳入
這邊有考慮到易讀性
還是說其實沒有影響 => 建議可以抽成 typescript enum 的寫法

Comment (shengyuan) [reply my-app/src/MonsterButton/MonsterButton.tsx:18]: 抽成 enum 後，可以在需要的地方引入。
好處是不會到處有特定的字串，在閱讀及管理上較為清楚。

其他專案的範例：

image

**Pull request closed without merging**